lightning/src/ln/functional_tests.rs

```diff
@@ -9992,3 +9992,129 @@ fn test_payment_with_custom_min_cltv_expiry_delta() {
 	do_payment_with_custom_min_final_cltv_expiry(true, false);
 	do_payment_with_custom_min_final_cltv_expiry(true, true);
 }
+
+#[test]
+fn test_disconnects_peer_awaiting_response_ticks() {
```

Comment:
I may have missed it, but going through the cases here, I don't see the specific one where receiving a `ChannelReestablish` or `RevokeAndACK` resets the counter to `None` if we have ticked, but less than `DISCONNECT_PEER_AWAITING_RESPONSE_TICKS`, and then receive the message.

Reply:
Ah yes, the very last bit. I see.